.circleci/config.yml

```diff
@@ -1,5 +1,106 @@
-version: 2
+version: 2.1
+
+# reusable 'executor' object for jobs
+executors:
+  go:
+    docker:
+      - image: circleci/golang:1.11.4
```

Comment:
Feel like we should have a makefile target that updates the go version across all of the files in the project at this point!

Reply:
We just have a hardcoded Golang version in CI / build / Docker related configuration files that requires updating when we want to bump globally. Some examples here but it is in README and elsewhere I believe. https://github.com/hashicorp/consul/pull/4782

Regardless not something for now.